api/graphql.go

```diff
@@ -0,0 +1,109 @@
+/*
+Gets factors that are used to build the rating for correctness of modules via
+GitHub's GraphQL API. getCorrectnessFactors function returns factors in the
+order of watchers, stargazers, totalCommits. A data type in the same form as
+the query structure is required to convert string to json. From json, the data
+is returned.
+
+--Use of GitHub token needs to be changed
+*/
+
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type CorrectnessFactors struct {
+	Data struct {
+		Repository struct {
+			StargazerCount int64
+			Watchers       struct {
+				TotalCount int64
+			}
+			DefaultBranchRef struct {
+				Target struct {
+					History struct {
+						TotalCount int64
+					}
+				}
+			}
+		}
+	}
+}
+
+func buildQuery(ownerName string, repoName string) (query map[string]string) {
+	var correctnessQuery = map[string]string{
+		"query": `
+		{
+			repository(owner:` + `"` + ownerName + `", name:` + `"` + repoName + `") {
+				name 
+				stargazerCount
+				watchers {
+					totalCount
+				}
+				defaultBranchRef {
+					target {
+						... on Commit {
+							history(first:0) {
+								totalCount
+							}
+						}
+					}
+				}
+			}
+		}`,
+	}
+
+	return correctnessQuery
```

Comment:
This is a function for the correctnessquery it looks like. I would rename the func to not just be "buildQuery". tis too generic